File: leetcode/lc-us/cpp/2325.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string decodeMessage(string key, string message) {
        unordered_map<char, char> M;
        char x = 'a';
        for (char c: key) {
            if (c != ' ') {
                if (!M.count(c)) M[c] = x++;
            }
        }
        string res;
        for (char c: message) {
            if (c == ' ') res.push_back(c);
            else res.push_back(M[c]);
        }
        return res;
    }
};

File: leetcode/lc-us/py/2325.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        d = {}
        letter = 'a'
        for c in key:
            if c != ' ' and c not in d:
                d[c] = letter
                letter = chr(ord(letter) + 1)
        res = ''
        for c in message:
            if c == ' ':
                res += ' '
            else:
                res += d[c]
        return res
